src/Service/PaymentStateDataService.php

```diff
@@ -0,0 +1,139 @@
+<?php declare(strict_types=1);
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2020 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <shopware@adyen.com>
+ */
+
+namespace Adyen\Shopware\Service;
+
+use Adyen\Shopware\Entity\PaymentStateData\PaymentStateDataEntity;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Psr\Log\LoggerInterface;
+
+class PaymentStateDataService
+{
+    /**
+     * @var EntityRepositoryInterface
+     */
+    protected $paymentStateDataRepository;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * PaymentStateDataService constructor.
+     * @param EntityRepositoryInterface $paymentStateDataRepository
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        EntityRepositoryInterface $paymentStateDataRepository,
+        LoggerInterface $logger
+    ) {
+        $this->paymentStateDataRepository = $paymentStateDataRepository;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param string $contextToken
+     * @param string $stateData
+     * @param string $origin
+     */
+    public function insertPaymentStateData(string $contextToken, string $stateData, string $origin): void
+    {
+
+        if (empty($contextToken) || empty($stateData)) {
+            $this->logger->error('No context token or state.data found, unable to save payment state.data');
+            exit;
```

Comment:
is the exit appropriate here?

Reply:
Changed to exception throw and added to PW-2836